cmd/notation/plugin/install.go

```diff
@@ -136,19 +136,18 @@ func install(command *cobra.Command, opts *pluginInstallOpts) error {
 		if pluginURL.Scheme != "https" {
 			return fmt.Errorf("failed to download plugin from URL: only the HTTPS scheme is supported, but got %s", pluginURL.Scheme)
 		}
-		tmpFile, err := os.CreateTemp("", notationPluginDownloadTmpFile)
+		tmpDir, err := os.MkdirTemp("", pluginDownloadTmpDir)
 		if err != nil {
-			return fmt.Errorf("failed to create temporary file required for downloading plugin: %w", err)
+			return fmt.Errorf("failed to create temporary directory required for downloading plugin: %w", err)
 		}
-		defer os.Remove(tmpFile.Name())
-		defer tmpFile.Close()
+		defer os.RemoveAll(tmpDir)
 		fmt.Printf("Downloading plugin from %s\n", opts.pluginSource)
-		err = notationplugin.DownloadPluginFromURL(ctx, opts.pluginSource, tmpFile)
+		tmpFile, err := notationplugin.DownloadPluginFromURL(ctx, opts.pluginSource, tmpDir)
```

Comment:
Here's a question: How do you know the filename of `tmpFile` is properly named as `notation-<plugin_name>[.exe]`?

Reply:
@shizhMSFT This is because if the plugin is a `single executable file` and it follows the notary project [spec](https://github.com/notaryproject/specifications/blob/main/specs/plugin-extensibility.md#installation), the `filename` has to be in `notation-{plugin-name}` format. If it doesn't, then Notation would fail the installation because we check the filename format before the install (in notation-go).
If the plugin is a `zip/tar.gz`, then the `filename` is not used/validated. Instead, files inside the `zip/tar.gz` are used/validated.

Reply:
@Two-Hearts What if I want to install `notation-my-plugin` (plugin name `my-plugin`) from `https://packages.contoso.com/notation/plugins/notation-my-plugin-linux-am64` where `notation-my-plugin-linux-am64` does not match `notation-my-plugin`? Is there a way to resolve it?

Reply:
> @Two-Hearts What if I want to install `notation-my-plugin` (plugin name `my-plugin`) from `https://packages.contoso.com/notation/plugins/notation-my-plugin-linux-am64` where `notation-my-plugin-linux-am64` does not match `notation-my-plugin`? Is there a way to resolve it?

The http response header `Content-Description filename` is checked first. If that's also not available, then one cannot directly install this plugin from URL using `notation plugin install`. They can still download the file onto file system and install from there though.